### PR TITLE
ref(apidocs): moved organization projects to drf spectacular

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
Was running through the new documentation tool and used Organization Projects as the guinea pig endpoint.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_projects.py [hunk 1]
```diff
@@ -1,29 +1,110 @@
...
+from sentry.apidocs.constants import RESPONSE_FORBIDDEN, RESPONSE_NOTFOUND, RESPONSE_UNAUTHORIZED
+from sentry.apidocs.parameters import CURSOR_QUERY_PARAM, GLOBAL_PARAMS
+from sentry.apidocs.utils import inline_sentry_response_serializer
 from sentry.models import Project, ProjectStatus, Team
 from sentry.search.utils import tokenize_query
 
 ERR_INVALID_STATS_PERIOD = "Invalid stats_period. Valid choices are '', '24h', '14d', and '30d'"
 
 
+class TeamResponseDict(TypedDict):
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_projects.py:23]: Have we thought about having shared serializer types for resources like team, projects and users which show up in multiple API responses?

Comment (AniketDas-Tekky) [reply src/sentry/api/endpoints/organization_projects.py:23]: I can move stuff together into a common lib. I'll move them after I get tests passing.

**Review by JoshFerge — CHANGES_REQUESTED** [review]
The ideal solution would be to type the serializer that the endpoint is utilizing, in this case `ProjectSummarySerializer`.

**Comment by JoshFerge** [comment]
can see `src/sentry/api/serializers/models/team.py` for how we type serializers. (only in endpoints where we don't have django serializers for responses did i use TypedDicts directly).

File: src/sentry/api/serializers/models/project.py [hunk 2]
```diff
@@ -475,6 +480,55 @@ def serialize(self, obj, attrs, user):
         return data
 
 
+##########
```
Comment (markstory) [inline src/sentry/api/serializers/models/project.py]: What are these comments for?

Comment (AniketDas-Tekky) [reply src/sentry/api/serializers/models/project.py]: I think you're looking at an older version? I don't see this line anymore

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**